src/infra/dex/paraswap/dto.rs

```diff
@@ -52,13 +51,20 @@ pub struct PriceQuery {
 
     /// The maximum price impact accepted (in percentage, 0-100)
     pub max_impact: u8,
+
+    /// The address of the signer.
+    pub user_address: H160,
+
+    /// A relative slippage tolerance denominated in bps.
+    pub slippage: u16,
 }
 
-impl PriceQuery {
+impl SwapQuery {
     pub fn new(
         config: &super::Config,
         order: &dex::Order,
         tokens: &auction::Tokens,
+        slippage: dex::Slippage,
```

Comment:
This could be taken by reference which would also get rid of a bunch of other changes need to have this available as a value here.

Reply:
https://github.com/gnosis/solvers/blob/f05a36eec420ed534d925e2fa85031b874413c10/src/infra/dex/paraswap/mod.rs#L114 requires it to be copied then.

Reply:
I think cloning this is fine to keep the API more consistent across different dexes. 👌